### ISSUE TITLE
Screen reader not detected on macOS

### ISSUE THREAD

**Issue by kieferrm** [microsoft/vscode#167627]
Testing #167159

Version: 1.74.0-insider
Commit: 8b9891739b703b50547f506a7e2bc9565e52beff
Date: 2022-11-29T07:07:14.672Z
Electron: 19.1.6
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin arm64 21.6.0
Sandboxed: Yes

Automatic screen reader detection doesn't work for me at all, i.e. in none of the scenarios listed in the test plan item. The only way to get in and out of screen reader mode is to use the command.

**Comment by deepak1556** [comment]
Synced with @kieferrm and confirmed that the issue also persists with stable on the same setup. Only difference between my setup is the OS version, I am on macOS ventura and the issue is seen on macOS monterey. @isidorn can you test on your device as well to validate this.

**Comment by isidorn** [comment]
I am on macOS ventura and for me this works just fine - both if I first start screen reader, or first start VS Code Insiders.
The only issue I see is that after reload screen reader is not detected - which should be covered by https://github.com/microsoft/vscode/issues/167583

fyi @meganrogge since I know you are also on macOS

**Comment by meganrogge** [comment]
On macOS Ventura, with Insider's already open, I expect to see the status bar entry when I turn VoiceOver on, but do not.

Maybe related - I had just turned off screen reader mode prior to this by answering the question `Are you using a screen reader to operate VS Code? No`.

https://user-images.githubusercontent.com/29464607/204815613-a89362f5-1ada-4ff3-ac64-a2a0d9ded73c.mov

**Comment by isidorn** [comment]
@meganrogge double check your settings.json. Ideally you should not have a `editor.accessibilitySupport` setting set. If you do and it is set to `false` that means already expressed you are not using a screen reader.

**Comment by meganrogge** [comment]
thanks @isidorn it doesn't work for me when I set that to `auto` either

**Comment by meganrogge** [comment]
When I toggle VoiceOver, this line does not get hit even though the value gets correctly updated in `accessibilityService`

https://github.com/microsoft/vscode/blob/e8e7adf4f26e27005e127d1d7390e5c1a3904a20/src/vs/workbench/browser/parts/editor/editorStatus.ts#L672

I think that's because we don't update the configuration value directly here

https://github.com/microsoft/vscode/blob/e8e7adf4f26e27005e127d1d7390e5c1a3904a20/src/vs/platform/accessibility/browser/accessibilityService.ts#L105-L114

So, we should forward the `_onDidChangeScreenReaderOptimized` event to `editStatus` I think

**Comment by meganrogge** [comment]
heard from @kieferrm that the value in his case is not set correctly. The issue I described is different as it's strictly a UI problem.

**Comment by meganrogge** [comment]
@deepak1556 I see that this feature is still being rolled out to stable. Does that mean we just wait or should I create an issue for Chromium?
![Screenshot 2022-11-30 at 3 43 45 PM](https://user-images.githubusercontent.com/29464607/204914450-824d5bf2-01f7-4051-be46-e0049cb7dba1.png)

**Comment by deepak1556** [comment]
Runtime issue has been resolved, leaving the issue open for https://github.com/microsoft/vscode/pull/167960

**Comment by meganrogge** [comment]
This is working for me, no need for my changes

**Comment by rzhao271** [comment]
There are still issues on macOS Monterey:
1. :bug: If one reloads Insiders while still keeping VoiceOver open, detection fails
2. :bug: If one starts VoiceOver first, detection fails
3. ✅ If one starts Insiders, and then starts VoiceOver, detection works

**Comment by deepak1556** [comment]
Thanks for testing! 1) and 2) are not regressions from this issue but behavior that has been existed before. I have opened separate issues to track these. In both the scenarios, the runtime puts the application into a11y mode, its just that our components to indicate the mode is not updated properly.
